###Prologue
this is my project log until I entered the dev-art project.

####2010

Back in 2010 I had the following idea:
The information stored in the human dna is represented by a nearly endless sequence of only four different letters: T, C, G and A. What if I interpret those four letters as drawing directions: T means one step up, C one step to the right, and so on. How will it look like? Are there emerging patterns? Could that idea even have scientific relevance?
Back then early experiments showed interesting results, however due to other obligations this idea moved out of my focus.

####February 2014
Heard about the devart project, and the old idea popped up again.

####Feb 21 14
+ started with research about genetics, mainly wikipedia: genome, dna, chromosome, nucleic bases
+ came across amino acids: amino acids seem to be defined as a sequence of three nucleic bases, there are around 24 amino acids symbolized by letters:
![Amino Acid code sun](http://upload.wikimedia.org/wikipedia/commons/7/70/Aminoacids_table.svg "Amino Acid code sun")

**Finding** applying my drawing rules did not lead to specific patters as the line combinations cannot clearly be mapped back to the amino acid symbols. -> I'll stick with my old main idea.

####Feb 23 14, tcg_i1_1

**Data** downloaded chromosome 1 of human from [http://hgdownload.soe.ucsc.edu/downloads.html](http://hgdownload.soe.ucsc.edu/downloads.html)

**Code** first sketch, render chromosome1
![chromosome 1, excerpt](../images/2014224_0.41chr1.jpg "chromosome 1, excerpt")

**Findings** 
+ some repetitions are visible. 
+ overall direction toward upper left corner. Means the sequence consists mostly of Ts and As.
+ did tests with different mappings of tcga to up-right-down-left. visual output differs if opposing chars are altered otherwise the picture is flipped or rotated.
-> while t-c-g-a  for up, right, down, left gives the most appealing result,  I stay with this config for now.
+ **research:** tcga is often referred to as base pairs , where *t matches with a, c with g*, so they are some kind of counter parts, therefore it seems logic to mape t & a and c & g  to opposite directions 

**data structure** 

+ the specs of .fa format can be found here: http://en.wikipedia.org/wiki/FASTA_format
+ the files consist of parts with only the letter N. N means "nucleic acid is unknown".  I ignore them.
+ upper and lowercase letter are found, I treat uppercase the same way as lower case letters. 
+ **TODO** find out more about the fasta file format


**presentation, exhibition**

whole image must be very large, but might have large void spaces..
idea: print quadratic tiles only if they are used by the visualisation, yields a superstructure well visible from a distance.

![sketch, raster system for exhibition space](../images/sketch1.jpg "sketch, raster system for exhibition space")

![sketch, tiled image on the wall](../images/sketch2.jpg "sketch, tiled image on the wall")

600 dpi = ca 200dots per centimeter, 1x1m = 20000x20000 pixels.. 


According to http://en.wikipedia.org/wiki/Chromosome, the human genome is divided into 22 chromosomes plus a X and Y chromosomes that defines the sex of the individuum. Chromosomes consist of genes, each gene is a sequence of nucleic acids. A nucleic acid is either represented by T, C, G or A.

####Feb 24 14, tcg_i2

**goal** render all chromosomes of the human genome. 

**strategy** render all chromosomes on separate images and then combine them manually

images are rendered with with a scaling of 0.05: means each pixel contains the information of 20 nucleic acids.

![chromosome 1](../images/chrom_tcg_i2/chr1.png "chromosome 1")
chromosome 1

![chromosome 2](../images/chrom_tcg_i2/chr2.png "chromosome 2")
chromosome 2

![chromosome 3](../images/chrom_tcg_i2/chr3.png "chromosome 3")
chromosome 3

![chromosome 4](../images/chrom_tcg_i2/chr4.png "chromosome 4")
chromosome 4

![chromosome 4, horse heads](../images/chrom_tcg_i2/chr4_detail.png "chromosome 4, horse heads")
chromosome 4, horse heads

![chromosome 5](../images/chrom_tcg_i2/chr5.png "chromosome 5")
chromosome 5

![chromosome 6](../images/chrom_tcg_i2/chr6.png "chromosome 6")
chromosome 6

![chromosome 7](../images/chrom_tcg_i2/chr7.png "chromosome 7")
chromosome 7

![chromosome 8](../images/chrom_tcg_i2/chr8.png "chromosome 8")
chromosome 8

![chromosome 9](../images/chrom_tcg_i2/chr9.png "chromosome 9")
chromosome 9

![chromosome 10](../images/chrom_tcg_i2/chr10.png "chromosome 10")
chromosome 10

![chromosome 10 diagonalis](../images/chrom_tcg_i2/chr10_detail.png "chromosome 10 diagonalis")
chromosome 10, diagonalis

![chromosome 11](../images/chrom_tcg_i2/chr11.png "chromosome 11")
chromosome 11

![chromosome 12](../images/chrom_tcg_i2/chr12.png "chromosome 12")
chromosome 12

![chromosome 13](../images/chrom_tcg_i2/chr13.png "chromosome 13")
chromosome 13

![chromosome 14](../images/chrom_tcg_i2/chr14.png "chromosome 14")
chromosome 14

![chromosome 15](../images/chrom_tcg_i2/chr15.png "chromosome 15")
chromosome 15

![chromosome 16](../images/chrom_tcg_i2/chr16.png "chromosome 16")
chromosome 16

![chromosome 17](../images/chrom_tcg_i2/chr17.png "chromosome 17")
chromosome 17

![chromosome 18](../images/chrom_tcg_i2/chr18.png "chromosome 18")
chromosome 18

![chromosome 19](../images/chrom_tcg_i2/chr19.png "chromosome 19")
chromosome 19

![chromosome 20](../images/chrom_tcg_i2/chr20.png "chromosome 20")
chromosome 20

![chromosome 21](../images/chrom_tcg_i2/chr21.png "chromosome 21")
chromosome 21

![chromosome 22](../images/chrom_tcg_i2/chr22.png "chromosome 22")
chromosome 22

![chromosome 22 cluster](../images/chrom_tcg_i2/chr22_details.png "chromosome 22 cluster")
chromosome 22 cluster

**observation** found interesting details in e.g. chromosome 4 and chromosome 10. some parts of the sequence seems to be duplicated, although they are not exact copies. Is this the same data deciphered twice, or is this a fuzzy repetition in the data? Its amazing how fast our eye finds those repeating structures. 

**how long would it take to find those repetitions algorithmically?**

####Feb 26.2014, tcg_i3
**goal** visualise the  human genome as one continuos string. how large is it?

![all chromosomes in one continuous sequence](../images/whole_genome.png "all chromosomes in one continuous sequence")

+ if this would be rendered in scale 1:1, means each nucleic acid would mean a movement by one pixel, **the shape would be approx. 1.2 million pixels tall and 0.35 million pixels wide.
+ *if this would be printed at 600dpi the image would be 2000 inches (ca. 51 meters) high......* 

####Mar 3 14 tcg_i3_3
**goal** some pixels get overdrawn many times. currently I draw with “solid” colours, so this information is lost. I want to switch to an “aquarell” colouring with transparency.

**problem** drawing with transparency to offscreen buffer did not work somehow.

**workaround** read and write directly to the pixels() array. As a side effect, this made the program much faster:)

code snippet: how to make an pixel a little bit darker
```
color c = context.pixels[p];

int r = (c >> 16) & 0xFF;  
int g = (c >> 8) & 0xFF;   
int b = c & 0xFF;  

r = int( float(r) * 0.99);
g = int( float(g) * 0.99);
b = int( float(b) * 0.99);

color new_c =  color(r,g,b); 
context.pixels[p] = new_c;
```
*observation* visuals get more interesting, have more depth.

![zoom1](../images/zoom1.png "zoom1")
![zoom2](../images/zoom2.png "zoom2")
![zoom3](../images/zoom3.png "zoom3")
![zoom4](../images/zoom4.png "zoom4")
![zoom5](../images/zoom5.png "zoom5")

**idea** *Use google maps to make the imagery accessible and zoomable!*
+ add aditional information, such as: "here ends chromosome 1", "this area codes for eye color"
+ "fly" to a specific point in the sequence
+ let people treat it as an unknown territory, where they can discover "landmarks" and name them. ("horse head", "diagonalis")


####Mar 5 14 tcg_i4
**research** there is a distinction between coding parts (genes) and non coding parts (trash dna?) in a chromosome. downloaded some annotation information and I use this to colour genes differently than non coding areas. hope to see that the repeating sequences are non-coding and the complicated structures are coding areas. 
TODO: talk to expert

**goal** distinguish genes from trash dna

![Genes Colored](../images/coloredGenes.png "Genes Colored")
red is Gene, grey is non coding area

**observation** although visually appealing, I wasn't able to see a correlation between shapes and colors. Hoped that the pictures show the repeating parts to be non-coding or something like that.



**concept** showed some screenshots to a friend, he said:”I want my own dna on the wall..”did some research, found this: 
http://www.forbes.com/sites/matthewherper/2014/01/14/the-1000-genome-arrives-for-real-this-time/ and http://en.wikipedia.org/wiki/$1,000_genome
**things seem to be closer than I thought. I change the project name to “This is you”**








**data**downloaded some datasets of some interesting  animals such as honey bee  etc… 
**Is the visualisation of the shimpansee more similar to the human than the visualisation of a honey bee?**
Problem: seems like the data I have for the human has a different format than the data of animals.. MRNA vs. DNA.. 
**TODO**find out about MRNA vs DNA









Interesting:

The $1000 Genome: Ethical and Legal Issues in Whole Genome Sequencing of Individuals 
http://www.utexas.edu/law/faculty/jrobertson/1000genome.pdf

I didnt know that it could be affordable for individuals to decipher their genome in the near future...
more about that here: http://en.wikipedia.org/wiki/$1,000_genome

I think I'll change the project title to "This is you" 

